source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -505,12 +597,26 @@ HttpConnectionManagerConfig::createCodec(Network::Connection& connection,
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
-void HttpConnectionManagerConfig::createFilterChain(Http::FilterChainFactoryCallbacks& callbacks) {
-  for (const Http::FilterFactoryCb& factory : filter_factories_) {
-    factory(callbacks);
+void HttpConnectionManagerConfig::createFilterChainForFactories(
+    Http::FilterChainFactoryCallbacks& callbacks, const FilterFactoriesList& filter_factories) {
+  for (const auto& filter_config_provider : filter_factories) {
+    auto config = filter_config_provider->config();
+    if (config.has_value()) {
+      config.value()(callbacks);
+      continue;
+    }
+
+    // If a filter config is missing after warming, inject a local reply with status 500.
```

Comment:
What if there are multiple filters with no config? This will get injected twice? Also what about if the missing config is an encoding filter?

I wonder if we should avoid this complexity now and just avoid this entirely by just setting a boolean and doing a direct 503 return in the HCM when headers are received? wdyt?

Reply:
Cutting it early in HCM is an approximation. If there's some telemetry filter ahead of a disabled filter, I think we may want to keep that filter active. I can create a singleton for the missing filter if space is a concern.

Reply:
Yeah that occurred to me, so It think doing it that this way makes sense, but it's kind of complicated. What about an encoding filter that is missing? This won't work. Personally I would keep it simple for now and do a follow up? Up to you.

Reply:
A missing encoding filter will be replaced by this decoding filter. The factory can be changed dynamically.

Reply:
Sorry I think my original comment wasn't clear. If you have multiple filters that have no config, you are going to add the below decoder filter multiple times. I think that will work but it's kind of strange. Can you make it only add once?

Reply:
Done.